src/core/parseHTML.js

```diff
@@ -15,7 +15,7 @@ define([
 // defaults to document
 // keepScripts (optional): If true, will include scripts passed in the html string
 jQuery.parseHTML = function( data, context, keepScripts ) {
-	if ( !data || typeof data !== "string" ) {
+	if ( typeof data !== "string" ) {
 		return null;
 	}
```

Comment:
Native behavior always casts to string except `null` (which is inconsistent between browsers): http://jsfiddle.net/evse952q/1/

Achieving that in jQuery would mean adopting my above `buildFragment` suggestion and casting (some) nullish values to string in `parseHTML`.